js/spotify.js

```diff
@@ -1,12 +1,283 @@
-// API Docs at: 
-// https://developer.spotify.com/web-api/search-item/
+$(function() {
+  getApp().initPage().listen();
+});
 
+function getApp() {
+  var searchTypes = defineSearchTypes();
+  var form = defineForm();
+  var message = defineMessage();
+  var results = defineResults();
+  var search = defineSearch();
 
-function searchByArtist(keyword) {
-  var url = 'https://api.spotify.com/v1/search?q='+keyword+'&type=artist';
-}
+  // App's API
+  return {
+    initPage: function() {
+      initPage();
+      return this;
+    },
+
+    listen: function() {
+      listen();
+      return this;
+    }
+  };
+
+  function defineSearchTypes() {
+    return ['artist', 'track', 'album', 'playlist'].map(function(name) {
+      return {
+        name: function() {
+          return name;
+        },
+
+        plural: function() {
+          return name + 's';
+        }
+      };
+    });
+  }
+
+  function defineForm() {
+    var element = $('#search');
+    var searchType = defineSearchType();
+    var keyword = defineKeyword();
+    var button = defineButton();
+    return {
+      element: function() {
+        return element;
+      },
+
+      searchType: function() {
+        return searchType;
+      },
+
+      keyword: function() {
+        return keyword;
+      },
+
+      button: function() {
+        return button;
+      }
```

Comment:
is there a reason the variables are separate from the functions here? you could just as well:

```js
return {
  element: $("#search") // ...etc
}
```